script/topic.js

```diff
@@ -44,171 +45,253 @@
-                    } else if (url.indexOf(llab.rootURL) == -1 && url.indexOf("..") == -1) {
-                        var slash = url[0] == "/" ? '' : '/';
-                        url = llab.rootURL + slash + url;
-                    }
-                    url += (url.indexOf('?') !== -1 ? '&' : '?') + llab.QS.stringify(query);
-                    num += 1;
-                    temp.attr({'href': url});
-                    item.append(temp);
-                } else {
-                    item.append(line.slice(sepIdx + 1));
+                topic_model = { type: 'topic', url: llab.topic, contents: [] };
+                topics.topics.push(topic_model);
+                section = { title: '', contents: [], type: 'section' };
+                topic_model.contents.push(section);
+            } else if (llab.isHeading(line)) {
+                headingType = llab.getKeyword(line, llab.topicKeywords.headings);
+                if (section.contents.length > 0) {
+                    section = { title: '', contents: [], type: 'section' };
+                    topic_model.contents.push(section);
                 }
-                topic.append(item);
+                section.title = llab.getContent(line)['text'];
+                section.headingType = headingType;
+            } else if (line[0] == "}") {
+                // shouldn't matter
+            } else if (llab.isInfo(line)) {
+                tag = llab.getKeyword(line, llab.topicKeywords.info);
+                indent = llab.indentLevel(line);
+                content = llab.getContent(line)['text']; // ?
+                // TODO: do we really need indentation now?
+                // if so, I think it should be added to the type
+                // and only if indentation levels != nested levels.
+                item = { type: tag, contents: content, indent: indent };
+                section.contents.push(item);
+            } else if (llab.isResource(line) || true) {
+                // FIXME: dumb way to handle lines without a known tag
+                // Shouldn't this just be an else case?
+                tag = llab.getKeyword(line, llab.topicKeywords.resources);
+                indent = llab.indentLevel(line);
+                content = llab.getContent(line);
+                item = { type: tag, indent: indent, contents: content.text,
+                         url: content.url };
+                section.contents.push(item);
             }
-        } else if (line.length == 1) {
-            learningGoal = false;
-            bigIdea = false;
+        } else if (line.length == 0) {
             raw = false;
-        } else if (raw) {
-            var raw_html = "";
-            while (line.length > 1 && line.slice[0] != "}") {
-                raw_html += " " + line;
+        }
+        if (raw) {
+            var raw_html = [];
+            text = llab.getContent(line)['text']; // in case they start the raw html on the same line
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>['text'] is better written in dot notation.